flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/SingleCheckpointBarrierHandler.java

```diff
@@ -235,6 +257,23 @@ int getNumOpenChannels() {
 		return numOpenChannels;
 	}
 
+	private CheckpointBarrier maybeTimeout(CheckpointBarrier barrier) {
+		CheckpointOptions options = barrier.getCheckpointOptions();
+		boolean shouldTimeout = (options.isTimeoutable()) && (
+			barrier.getId() == timeoutedBarrierId ||
+			(System.currentTimeMillis() - barrier.getTimestamp()) > options.getAlignmentTimeout());
```

Comment:
We discussed offline two options of switching a checkpoint from aligned to unaligned:
1. after some time since the start of a checkpoint on JM
1. after some time since the start of alignment on this subtask

(in either case, all the downstream tasks would proceed in UC mode with this checkpoint after it's timed out)

The latter gives more control, i.e. the ability to address long alignment in isolation, but not long barrier travel time (e.g. long sync phase or just long pipeline).

We couldn't agree on which one would be more convenient for the users.

@NicoK, do you have an opinion on that?

Or maybe we should provide both?

Reply:
After a discussion with @NicoK, @sjwiesman and @alpinegizmo we decided to:
1. Decide to timeout based on the alignment start time
1. By default, propagate this decision downstream; provide an option to disable propagation
1. In the UI, show checkpoint type for each subtask; on a checkpoint level display unaligned if at least one subtask did UC
1. Consider renaming `alignment timeout` option to  `subtask alignment timeout` 

Considerations:
- the overhead of UC (persisting channels) should ideally be localized
- the less global the decision is, the more difficult it might be to debug UC-related issues
- In a common scenario, backpressure comes from sinks; buffers will be full, so disabling propagation doesn't make a difference